### PR TITLE
Migrate to slack notifications

### PR DESCRIPTION
**Pull request by philipbalinov**
Migrate away from Discord and to Slack notifications 

Notifications are going to #mondoo-ops via the Mondoo App

### CODE REVIEW

File: .github/workflows/check_gpg_key_not_expired.yml [hunk 1]
```diff
@@ -13,6 +13,41 @@ jobs:
...
+                  ]
+                }
+              ]
+            }
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+          # C07R9GSGKEU == #mondoo-ops
+          SLACK_BOT_CHANNEL_ID: "C07R9GSGKEU"
+          WORKFLOW_RUN_URL: ${{ github.event.workflow_run.html_url }}
+          WORKFLOW_RUN_NAME: ${{ github.event.workflow_run.name }}
```
Comment (czunker) [inline .github/workflows/check_gpg_key_not_expired.yml:48]: Perhaps set these on the job level, so we don't have them twice?

File: .github/workflows/check_gpg_key_not_expired.yml [hunk 2]
```diff
@@ -32,11 +67,39 @@ jobs:
...
           bash test/scripts/cert-harmony.sh
-          
-      - name: Discord notification
-        uses: Ilshidur/action-discord@0.3.2
-        if: failure()
-        env:
-          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+
+      - uses: slackapi/slack-github-action@v1.27.0
+        if: always()
```
Comment (czunker) [inline .github/workflows/check_gpg_key_not_expired.yml:72]: Is this was you want?

IIRC, `always()` was counterintuitive to me. In some places, I had to use `failure() || success()`.

File: .github/workflows/test_install_sh.yml [hunk 3]
```diff
@@ -33,6 +33,41 @@ jobs:
...
+                  ]
+                }
+              ]
+            }
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+          # C07R9GSGKEU == #mondoo-ops
+          SLACK_BOT_CHANNEL_ID: "C07R9GSGKEU"
+          WORKFLOW_RUN_URL: ${{ github.event.workflow_run.html_url }}
+          WORKFLOW_RUN_NAME: ${{ github.event.workflow_run.name }}
```
Comment (czunker) [inline .github/workflows/test_install_sh.yml:68]: Same here, Perhaps putting this on the job level makes life easier in the future.

File: .github/workflows/check_gpg_key_not_expired.yml (hunk 2 above)
Comment (philipbalinov) [reply .github/workflows/check_gpg_key_not_expired.yml:72]: yes, I want the action to update slack if the job is cancelled too

File: .github/workflows/check_gpg_key_not_expired.yml (hunk 1 above)
Comment (philipbalinov) [reply .github/workflows/check_gpg_key_not_expired.yml]: cleaned those vars up as they were actually not needed, and moved the remaining to the job level

**Review by czunker — APPROVED** [review]
Thanks @philipbalinov

**Pull request merged**